src/inet/IPAddress.h

```diff
@@ -201,6 +201,9 @@ class DLL_EXPORT IPAddress
      *  address in network byte order.
      */
     uint32_t Addr[4];
+#if CHIP_SYSTEM_CONFIG_USE_LWIP && LWIP_IPV6_SCOPES
+    uint8_t Zone = IP6_NO_ZONE;
```

Comment:
Does LwIP use zones for non-link-local bits?

Because for link-local, PeerAddress is supposed to have an interface ID already, which is what the zone bits are, no?  Is that not being handled correctly in our LwIP integration?  Because if that's what's going on, the right fix is to fix it there, not to add things to IPAddress that are also duplicated-but-unused-it's-a-trap in PeerAddress.